File: apps/sak/src/main/kotlin/no/nav/gjenlevende/bs/sak/behandling/Behandling.kt
package no.nav.gjenlevende.bs.sak.behandling

import no.nav.gjenlevende.bs.sak.felles.sporbar.Sporbar
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Embedded
import org.springframework.data.relational.core.mapping.Table
import java.util.UUID

@Table("behandling")
data class Behandling(
    @Id
    val id: UUID = UUID.randomUUID(),
    val fagsakId: UUID,
    val forrigeBehandlingId: UUID? = null,
    val status: BehandlingStatus,
    val resultat: BehandlingResultat,
    @Embedded(onEmpty = Embedded.OnEmpty.USE_EMPTY)
    val sporbar: Sporbar = Sporbar(),
)

enum class BehandlingStatus {
    OPPRETTET,
    UTREDES,
    FATTER_VEDTAK,
    IVERKSETTER_VEDTAK,
    FERDIGSTILT,
}

enum class BehandlingResultat {
    INNVILGET,
    OPPHØRT,
    AVSLÅTT,
    IKKE_SATT,
    HENLAGT,
}

File: apps/frontend/app/types/behandling.ts
export interface Behandling {
  id: string;
  fagsakId: string;
  forrigeBehandlingId?: string;
  status: "OPPRETTET" | "UTREDES" | "FATTER_VEDTAK" | "IVERKSETTER_VEDTAK" | "FERDIGSTILT";
  sistEndret: string;
  sistEndretAv: string;
  opprettet: string;
  opprettetAv: string;
  resultat: "INNVILGET" | "OPPHØRT" | "AVSLÅTT" | "IKKE_SATT" | "HENLAGT";
}
